hpy/devel/include/hpy/hpydef.h

```diff
@@ -99,6 +99,10 @@ typedef struct {
     HPyFunc_Capsule_Destructor impl;
 } HPyCapsule_Destructor;
 
+typedef struct {
+    cpy_vectorcallfunc cpy_trampoline;
+    HPyFunc_vectorcallfunc impl;
+} HPyVectorcall;
```

Comment:
Because in the _manual_ case, you need to include this struct (I've renamed it to `HPyCallFunction`) in the type's struct like this:
```
typedef struct {
    int member0;
    int member1;
    // ...
    HPyCallFunction callfunc;
   // ...
}
```
In this case, the struct shouldn't be larger than necessary which would be the case if we use `HPyDef`. For consistency, I'm also using `HPyCallFunction *` in `HPyVectorcall_Set` (name of this function is subject to be changed). We could use a slot definition there but that seems to be more confusing.

Reply:
ok, got it and it makes a lot of sense.
The only drawback which I see is that by doing this we are wasting a pointer "forever": `HPyCallFunction` contains a pointer to the cpy_trampoline, which is needed *right now* to support CPython, but it's not needed at all by alternative implementations and it might not be needed even by CPython itself in the future (in case they decide to support hpy natively).

Note that it's slightly different than the `HPySlot` case, because `HPySlot`s are in fixed number (and stored in static data), while this impose an extra cost to the runtime objects.

However, I don't really know how to solve this issue, so I suppose it is fine to keep as is for now, and see whether we can improve it later

Reply:
Right, that's a good point. We don't really need the both function pointers in general.
I think about how we can reduce that to void `sizeof(void *)` slot but yes, let's merge this PR and do it in a follow-up (since the PR is already pretty large).